app/api/queue.v2/infrastructure/MemoryQueueAdapter.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable max-classes-per-file */
```

Comment:
I guess you meant "production", because it in fact has very practical purposes :P It is used for tests and as a specification for the semantics of the interface it implements. I didn't put it into the tests because I was afraid that it was modified out of the semantics to fulfill some testing need in the future. But I guess all of that can be mitigated with a comment.